src/SqlPersistence/Config/SqlDialect_Oracle.cs

```diff
@@ -12,6 +13,8 @@ public abstract partial class SqlDialect
         /// </summary>
         public partial class Oracle : SqlDialect
         {
+            volatile PropertyInfo bindByNameProperty;
```

Comment:
And it can't be optimized into a static constructor or something like that because it has to wait for an OracleCommand object to pass through.

I'm not an expert on the volatile keyword by any means. Is something like this better, attempting to ensure that the instance of the PropertyInfo you're trying to deal with is always a local variable?

```
var localProperty = bindByNameProperty;
if(localProperty == null)
{
    localProperty = command.GetType().GetProperty("BindByName");
    bindByNameProperty = localProperty;
}
localProperty.SetValue(command, true);
```